README.md

```diff
@@ -1,724 +1,1034 @@
+Added Sales personnel. Logging all known personnel
+Denise is in Sales department
+Erwing is in Sales department
+Faith is in Sales department
+Alice is in IT department
+Clark is in IT department
+Bob is in IT department
 ```
 
-MyGlobalSerializer is your implementation of StreamSerializer interface. You do not need to write the getHazelcastTypeId function for your objects which are handled by MyGlobalSerializer class.
+You will see this time we add only the sales employees but we get the list all known employees including the ones in IT.
+That is because our map lives in the cluster and no matter which client we use, we can access the whole map.
 
-# Invocation And Retry Mechanism
-The client works by sending requests to the members in the cluster and processes the reponses. This is called a client invocation to the cluster. If any problem occurs during an invocation, the following rules apply to decide if the invocation should be retried:
+## 1.6. Code Samples
 
-- If the problem is an IOException, which means that the request could not even be sent to the cluster member, e.g., due to a network connection problem, then all such requests can be retried.
-- If the invocation was sent to the cluster and the member returned an exception to the client, the client will retry the invocation only if the operation is a retryable idempotent operation. For example, the IMap::get is idempotent and can be retried but IMap::put is not idempotent.
+See Hazelcast C++ [code samples](https://github.com/hazelcast/hazelcast-cpp-client/tree/master/examples) for more examples.
 
-When invocation is being retried, the client may wait some time before it retries again. This retry wait time is configurable with this configuration property:
+You can also see the Hazelcast C++ [API Documentation](https://docs.hazelcast.org/docs/clients/cpp/3.10.1/html/).
 
-```
-config.setProperty("hazelcast.client.invocation.retry.pause.millis", "500");
-```
+# 2. Features
 
-The default retry wait time is 1 second.
+Hazelcast C++ client supports the following data structures and features:
 
-Of course, this retry does not continue indefinitely but finishes when the invocation timeout occurs. This duration is configurable with this configuration property:
+* Map
+* Queue
+* Set
+* List
+* MultiMap
+* Replicated Map
+* Ringbuffer
+* Reliable Topic
+* Lock
+* Semaphore
+* Atomic Long
+* CRDT PN Counter
+* Flake Id Generator
+* Event Listeners
+* Entry Processor
+* Query (Predicates)
+* Paging Predicate
+* Built-in Predicates
+* Listener with Predicate
+* Near Cache Support
+* Programmatic Configuration
+* Fail Fast on Invalid Configuration
+* SSL Support (requires Enterprise server)
+* Authorization
+* Smart Client
+* Unisocket Client
+* Lifecycle Service
+* IdentifiedDataSerializable Serialization
+* Portable Serialization
+* Custom Serialization
+* Global Serialization
 
-```
-config.setProperty("hazelcast.client.invocation.timeout.seconds", "30");
-```
+# 3. Configuration Overview
```

Comment:
I left as it is since we will add the xml config also. Hence, it may thought as for future. If you think we go with your suggestion for now and change later, i am also OK with that.